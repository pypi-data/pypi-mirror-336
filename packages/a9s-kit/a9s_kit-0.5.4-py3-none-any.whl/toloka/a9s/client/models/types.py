from typing import NewType
from uuid import UUID

AnnotationId = NewType('AnnotationId', str)
AnnotationEditId = NewType('AnnotationEditId', str)
AnnotationMetricId = NewType('AnnotationMetricId', str)
BatchId = NewType('BatchId', str)
GroundTruthOutputValueId = NewType('GroundTruthOutputValueId', str)
GroundTruthBucketId = NewType('GroundTruthBucketId', str)
GroundTruthId = NewType('GroundTruthId', str)
GroundTruthConfigId = NewType('GroundTruthConfigId', str)
MoneyConfigId = NewType('MoneyConfigId', UUID)
MoneyConfigVersionId = NewType('MoneyConfigVersionId', UUID)
ProjectId = NewType('ProjectId', str)
PipelineId = NewType('PipelineId', str)
QualificationId = NewType('QualificationId', UUID)
QualificationFilterId = NewType('QualificationFilterId', UUID)
UserQualificationId = NewType('UserQualificationId', UUID)
PipelineConfigId = NewType('PipelineConfigId', UUID)
PipelineInstanceId = NewType('PipelineInstanceId', UUID)
ProgressLogId = NewType('ProgressLogId', UUID)
AnnotationGroupId = NewType('AnnotationGroupId', str)
AsyncQualityConfigId = NewType('AsyncQualityConfigId', str)
QualityConfigId = NewType('QualityConfigId', str)
RestrictionId = NewType('RestrictionId', str)
StatusWorkflowConfigId = NewType('StatusWorkflowConfigId', str)
TenantId = NewType('TenantId', str)
AnnotationProcessId = NewType('AnnotationProcessId', str)
ResetId = NewType('ResetId', str)
