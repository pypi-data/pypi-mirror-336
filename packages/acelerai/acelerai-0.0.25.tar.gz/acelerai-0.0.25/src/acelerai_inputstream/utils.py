from datetime import date, datetime
from enum import Enum
import hashlib
import json
import os
from uuid import UUID
import msgpack
from decimal import Decimal
from dateutil import parser

import logging
logger = logging.getLogger("AcelerUtil")

def custom_encoder(obj):
    if isinstance(obj, datetime):
        return obj.isoformat()
        
    if isinstance(obj, Decimal):
        return float(obj)  # Serializar Decimal como flotante
    raise TypeError(f"Object of type {type(obj).__name__} is not serializable")


def decode_datetime(obj):
    for key, value in obj.items():
        if isinstance(value, str):
            try:
                #caso para date y datetime
                if 'T' in value and '-' in value and ':' in value and len(value) < 40:
                    obj[key] = datetime.fromisoformat(value)
                elif '-' in value and len(value) < 11:
                    obj[key] = parser.parse(value) #.date()
                #obj[key] = datetime.fromisoformat(value)  # Deserializar datetime
            except ValueError:
                pass
    return obj


def load_full_object(file_path):
    """Carga completamente el objeto desde un archivo MessagePack en memoria."""
    try:
        with open(file_path, "rb") as file:
            # Cargar todos los registros en memoria como una lista
            unpacker = msgpack.Unpacker(file, raw=False, object_hook=decode_datetime)
            data = [record for record in unpacker]  # Deserializar todos los registros
        return data
    except Exception as e:
        logger.error(f"Error al cargar el archivo: {e}", exc_info=True)
        return None
    

class CustomJSONEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, Enum):
            return obj.value
        
        elif isinstance(obj, datetime):
            return obj.isoformat()
        
        elif isinstance(obj, date):
            #logger.debug(f"Date: {obj}")
            #return datetime.strptime(obj, "%Y-%m-%d").date()
            return obj.isoformat()
        
        elif isinstance(obj, UUID):
            return str(obj)
        else:
            return super().default(obj)


class CustomJsonDecoder(json.JSONDecoder):
    def __init__(self, *args ,**kargs):
        json.JSONDecoder.__init__(self, object_hook=self.object_hook, *args, **kargs)

    def object_hook(self, obj:dict):
        for k, v in obj.items():
            if isinstance(v, str) and 'T' in v and '-' in v and ':' in v and len(v) < 40:
                try:
                    dv = parser.parse(v)
                    dt = dv.replace(tzinfo=None)
                    obj[k] = dt
                except:
                    pass
            elif isinstance(v, str) and '-' in v and len(v) < 11:
                try:
                    obj[k] = parser.parse(v).date()
                except:
                    pass
        return obj



