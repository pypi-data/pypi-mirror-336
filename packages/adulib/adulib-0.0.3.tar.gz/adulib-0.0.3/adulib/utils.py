# AUTOGENERATED! DO NOT EDIT! File to edit: ../nbs/api/utils.ipynb.

# %% auto 0
__all__ = ['check_mutual_exclusivity']

# %% ../nbs/api/utils.ipynb 4
def check_mutual_exclusivity(*args, check_falsy=True):
    """
    Check if only one of the arguments is falsy (or truthy, if check_falsy is False).
    """
    if check_falsy:
        return sum(bool(x) for x in args) == 1
    else:
        return sum(bool(x) for x in args) == 1
