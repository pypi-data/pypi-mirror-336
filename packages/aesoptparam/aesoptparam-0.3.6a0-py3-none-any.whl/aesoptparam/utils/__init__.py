from .html_repr import display_json_data
from .json_utils import read_json, write_json


def copy_param(param, exclude=None, update=None):
    """Copy a Parameter, either a standalone `Parameter` (`AESOptNumber`, `AESOptArray`, ..) or from a `Parametrized` class (using `par_class.param.param_name`)

    Parameters
    ----------
    param : `Parameter`
        Parameter instance
    exclude : list, optional
        List of entries to exclude, by default None
    update : dict, optional
        Dict with values to update, by default None

    Returns
    -------
    `Parameter`
        Copy of parameter instance. Optionally without entries from `exclude` or updated values from `update`
    """
    if exclude is None:
        exclude = []
    if update is None:
        update = dict()
    kwargs = dict()
    for name in param._slot_defaults:
        val = getattr(param, name)
        if not name in exclude and val is not None:
            kwargs[name] = val
    for name, val in update.items():
        kwargs[name] = val

    return type(param)(**kwargs)


def copy_param_ref(param, ref, exclude=None, update=None):
    """Similar to `copy_param` but adding "default" to `exclude` and "default_ref" to `update` with value of `ref`

    Parameters
    ----------
    param : `Parameter`
        Parameter instance
    ref : str
        String with the reference path
    exclude : list, optional
        List of entries to exclude, by default None
    update : dict, optional
        Dict with values to update, by default None

    Returns
    -------
    `Parameter`
        Copy of parameter instance. Optionally without entries from `exclude` or updated values from `update`
    """
    if exclude is None:
        exclude = []
    if update is None:
        update = dict()
    if not "default" in exclude:
        exclude.append("default")
    update["default_ref"] = ref
    return copy_param(param, exclude, update)
