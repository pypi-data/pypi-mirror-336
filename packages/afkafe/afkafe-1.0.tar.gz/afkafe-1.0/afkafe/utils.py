from typing import Any


def default_log_config(verbose: bool) -> dict[str, Any]:
    """
    Setup default config. for dictConfig.

    :param verbose: level: DEBUG if True, INFO if False
    :return: dict suitable for ``logging.config.dictConfig``
    """
    log_level = "DEBUG" if verbose else "INFO"
    return {
        "version": 1,
        "disable_existing_loggers": False,
        "handlers": {
            "afkafe.standard": {
                "level": log_level,
                "class": "logging.StreamHandler",
                "formatter": "afkafe.standard",
            }
        },
        "formatters": {
            "afkafe.standard": {
                "format": "%(asctime)s: %(message)s",
                "datefmt": "%H:%M:%S",
            }
        },
        "loggers": {"afkafe": {"handlers": ["afkafe.standard"], "level": log_level}},
    }
