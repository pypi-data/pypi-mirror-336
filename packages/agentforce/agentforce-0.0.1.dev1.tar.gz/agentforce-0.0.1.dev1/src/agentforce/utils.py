import base64
import logging
import os

logger = logging.getLogger(__name__)


def image_to_base64_data_uri(file_path):
    with open(file_path, "rb") as img_file:
        base64_data = base64.b64encode(img_file.read()).decode("utf-8")
        return f"data:image/png;base64,{base64_data}"


def llama_cpp_image_handler(image_url_or_path: str):
    if image_url_or_path.startswith("http"):
        return image_url_or_path
    else:
        if not os.path.exists(image_url_or_path):
            raise ValueError(f"Path {image_url_or_path} does not exist")
        return image_to_base64_data_uri(image_url_or_path)
