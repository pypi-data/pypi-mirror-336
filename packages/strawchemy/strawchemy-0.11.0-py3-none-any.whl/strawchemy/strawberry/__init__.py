from __future__ import annotations

from strawchemy.types import DefaultOffsetPagination

from ._field import StrawchemyField
from ._instance import ModelInstance
from ._utils import default_session_getter

__all__ = ("DefaultOffsetPagination", "ModelInstance", "StrawchemyField", "default_session_getter")
