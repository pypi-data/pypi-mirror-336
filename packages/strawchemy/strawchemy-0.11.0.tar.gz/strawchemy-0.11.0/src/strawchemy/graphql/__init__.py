from __future__ import annotations

from .factory import DistinctOnFieldsDTOFactory, FilterDTOFactory, OrderByDTO, OrderByEnum
from .filters import GenericComparison, GraphQLFilter, JSONComparison, PostgresArrayComparison, TextComparison

__all__ = (
    "DistinctOnFieldsDTOFactory",
    "FilterDTOFactory",
    "GenericComparison",
    "GraphQLFilter",
    "JSONComparison",
    "OrderByDTO",
    "OrderByEnum",
    "PostgresArrayComparison",
    "TextComparison",
)
