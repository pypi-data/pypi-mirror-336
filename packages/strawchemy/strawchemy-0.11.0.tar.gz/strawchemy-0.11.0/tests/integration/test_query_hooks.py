from __future__ import annotations

from typing import TYPE_CHECKING

import pytest
from strawchemy import StrawchemyAsyncRepository, StrawchemySyncRepository

import strawberry
from tests.typing import AnyQueryExecutor
from tests.utils import maybe_async

from .types import ColorWithFilteredFruit, FilteredFruitType, FruitTypeWithDescription, OrderedFruitType, strawchemy
from .typing import RawRecordData

if TYPE_CHECKING:
    from syrupy.assertion import SnapshotAssertion

    from .fixtures import QueryTracker

pytestmark = [pytest.mark.integration]


@strawberry.type
class AsyncQuery:
    fruits: list[FruitTypeWithDescription] = strawchemy.field(repository_type=StrawchemyAsyncRepository)
    fruits_paginated: list[FruitTypeWithDescription] = strawchemy.field(
        repository_type=StrawchemyAsyncRepository, pagination=True
    )
    filtered_fruits: list[FilteredFruitType] = strawchemy.field(repository_type=StrawchemyAsyncRepository)
    filtered_fruits_paginated: list[FilteredFruitType] = strawchemy.field(
        repository_type=StrawchemyAsyncRepository, pagination=True
    )
    ordered_fruits: list[OrderedFruitType] = strawchemy.field(repository_type=StrawchemyAsyncRepository)
    ordered_fruits_paginated: list[OrderedFruitType] = strawchemy.field(
        repository_type=StrawchemyAsyncRepository, pagination=True
    )
    colors: list[ColorWithFilteredFruit] = strawchemy.field(repository_type=StrawchemyAsyncRepository)


@strawberry.type
class SyncQuery:
    fruits: list[FruitTypeWithDescription] = strawchemy.field(repository_type=StrawchemySyncRepository)
    fruits_paginated: list[FruitTypeWithDescription] = strawchemy.field(
        repository_type=StrawchemySyncRepository, pagination=True
    )
    filtered_fruits: list[FilteredFruitType] = strawchemy.field(repository_type=StrawchemySyncRepository)
    filtered_fruits_paginated: list[FilteredFruitType] = strawchemy.field(
        repository_type=StrawchemySyncRepository, pagination=True
    )
    ordered_fruits: list[OrderedFruitType] = strawchemy.field(repository_type=StrawchemySyncRepository)
    ordered_fruits_paginated: list[OrderedFruitType] = strawchemy.field(
        repository_type=StrawchemySyncRepository, pagination=True
    )
    colors: list[ColorWithFilteredFruit] = strawchemy.field(repository_type=StrawchemySyncRepository)


@pytest.fixture
def sync_query() -> type[SyncQuery]:
    return SyncQuery


@pytest.fixture
def async_query() -> type[AsyncQuery]:
    return AsyncQuery


@pytest.mark.parametrize("fruits_query", ["fruits", "fruitsPaginated"])
@pytest.mark.snapshot
async def test_load_columns_hook(
    fruits_query: str,
    any_query: AnyQueryExecutor,
    raw_fruits: RawRecordData,
    query_tracker: QueryTracker,
    sql_snapshot: SnapshotAssertion,
) -> None:
    result = await maybe_async(any_query(f"{{ {fruits_query} {{ description }} }}"))

    assert not result.errors
    assert result.data
    assert result.data[fruits_query] == [
        {"description": f"The {raw_fruits[0]['name']} is {', '.join(raw_fruits[0]['adjectives'])}"},
        {"description": f"The {raw_fruits[1]['name']} is {', '.join(raw_fruits[1]['adjectives'])}"},
        {"description": f"The {raw_fruits[2]['name']} is {', '.join(raw_fruits[2]['adjectives'])}"},
        {"description": f"The {raw_fruits[3]['name']} is {', '.join(raw_fruits[3]['adjectives'])}"},
        {"description": f"The {raw_fruits[4]['name']} is {', '.join(raw_fruits[4]['adjectives'])}"},
    ]

    assert query_tracker.query_count == 1
    assert query_tracker[0].statement_formatted == sql_snapshot


@pytest.mark.parametrize("fruits_query", ["fruits", "fruitsPaginated"])
async def test_empty_query_hook(fruits_query: str, any_query: AnyQueryExecutor) -> None:
    result = await maybe_async(any_query(f"{{ {fruits_query} {{ emptyQueryHook }} }}"))

    assert not result.errors
    assert result.data
    assert len(result.data[fruits_query]) == 5
    assert result.data[fruits_query] == [{"emptyQueryHook": "success"} for _ in range(5)]


@pytest.mark.parametrize("query", ["filteredFruits", "filteredFruitsPaginated"])
@pytest.mark.snapshot
async def test_custom_query_hook_where(
    query: str, any_query: AnyQueryExecutor, query_tracker: QueryTracker, sql_snapshot: SnapshotAssertion
) -> None:
    result = await maybe_async(any_query(f"{{ {query} {{ name }} }}"))

    assert not result.errors
    assert result.data
    assert len(result.data[query]) == 1
    assert result.data[query] == [{"name": "Apple"}]

    assert query_tracker.query_count == 1
    assert query_tracker[0].statement_formatted == sql_snapshot


@pytest.mark.parametrize("query", ["orderedFruits", "orderedFruitsPaginated"])
@pytest.mark.snapshot
async def test_custom_query_hook_order_by(
    query: str, any_query: AnyQueryExecutor, query_tracker: QueryTracker, sql_snapshot: SnapshotAssertion
) -> None:
    result = await maybe_async(any_query(f"{{ {query} {{ name }} }}"))

    assert not result.errors
    assert result.data
    assert len(result.data[query]) == 5
    assert result.data[query] == [
        {"name": "Apple"},
        {"name": "Banana"},
        {"name": "Orange"},
        {"name": "Strawberry"},
        {"name": "Watermelon"},
    ]

    assert query_tracker.query_count == 1
    assert query_tracker[0].statement_formatted == sql_snapshot


@pytest.mark.snapshot
async def test_query_hook_on_type(
    any_query: AnyQueryExecutor, query_tracker: QueryTracker, sql_snapshot: SnapshotAssertion
) -> None:
    result = await maybe_async(any_query("{ colors { fruits { name } } }"))

    assert not result.errors
    assert result.data
    assert len(result.data["colors"]) == 1
    assert result.data["colors"] == [{"fruits": [{"name": "Apple"}]}]

    assert query_tracker.query_count == 1
    assert query_tracker[0].statement_formatted == sql_snapshot
