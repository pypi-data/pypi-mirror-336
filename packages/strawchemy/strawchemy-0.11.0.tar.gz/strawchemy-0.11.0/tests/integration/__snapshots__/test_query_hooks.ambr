# serializer version: 1
# name: test_always_load_column_hook[session-tracked-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM fruit AS fruit
  '''
# ---
# name: test_always_load_column_hook[session-tracked-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM fruit AS fruit
  '''
# ---
# name: test_always_load_column_hook[session-tracked-fruits-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM fruit AS fruit
  '''
# ---
# name: test_always_load_column_hook[session-tracked-fruits-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM fruit AS fruit
  '''
# ---
# name: test_always_load_column_hook[session-tracked-fruits-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM fruit AS fruit
  '''
# ---
# name: test_always_load_column_hook[session-tracked-fruitsPaginated-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM
    (SELECT fruit.id AS id,
            fruit.name AS name,
            fruit.adjectives AS adjectives
     FROM fruit AS fruit
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  LIMIT :param_3
  '''
# ---
# name: test_always_load_column_hook[session-tracked-fruitsPaginated-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM
    (SELECT fruit.id AS id,
            fruit.name AS name,
            fruit.adjectives AS adjectives
     FROM fruit AS fruit
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  LIMIT :param_3
  '''
# ---
# name: test_always_load_column_hook[session-tracked-fruitsPaginated-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM
    (SELECT fruit.id AS id,
            fruit.name AS name,
            fruit.adjectives AS adjectives
     FROM fruit AS fruit
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  LIMIT :param_3
  '''
# ---
# name: test_always_load_column_hook[session-tracked-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM fruit AS fruit
  '''
# ---
# name: test_custom_query_hook[session-tracked-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  WHERE fruit.name = :name_1
  '''
# ---
# name: test_custom_query_hook[session-tracked-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  WHERE fruit.name = :name_1
  '''
# ---
# name: test_custom_query_hook[session-tracked-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  WHERE fruit.name = :name_1
  '''
# ---
# name: test_custom_query_hook[session-tracked-userFruits-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  WHERE fruit.name = :name_1
  '''
# ---
# name: test_custom_query_hook[session-tracked-userFruits-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  WHERE fruit.name = :name_1
  '''
# ---
# name: test_custom_query_hook[session-tracked-userFruits-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  WHERE fruit.name = :name_1
  '''
# ---
# name: test_custom_query_hook[session-tracked-userFruitsPaginated-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM
    (SELECT fruit.name AS name,
            fruit.id AS id
     FROM fruit AS fruit
     WHERE fruit.name = :name_1
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  WHERE fruit.name = :name_2
  LIMIT :param_3
  '''
# ---
# name: test_custom_query_hook[session-tracked-userFruitsPaginated-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM
    (SELECT fruit.name AS name,
            fruit.id AS id
     FROM fruit AS fruit
     WHERE fruit.name = :name_1
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  WHERE fruit.name = :name_2
  LIMIT :param_3
  '''
# ---
# name: test_custom_query_hook[session-tracked-userFruitsPaginated-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM
    (SELECT fruit.name AS name,
            fruit.id AS id
     FROM fruit AS fruit
     WHERE fruit.name = :name_1
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  WHERE fruit.name = :name_2
  LIMIT :param_3
  '''
# ---
# name: test_custom_query_hook_order_by[session-tracked-orderedFruits-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  ORDER BY fruit.name ASC
  '''
# ---
# name: test_custom_query_hook_order_by[session-tracked-orderedFruits-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  ORDER BY fruit.name ASC
  '''
# ---
# name: test_custom_query_hook_order_by[session-tracked-orderedFruits-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  ORDER BY fruit.name ASC
  '''
# ---
# name: test_custom_query_hook_order_by[session-tracked-orderedFruitsPaginated-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM
    (SELECT fruit.name AS name,
            fruit.id AS id
     FROM fruit AS fruit
     ORDER BY fruit.name ASC
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  ORDER BY fruit.name ASC
  LIMIT :param_3
  '''
# ---
# name: test_custom_query_hook_order_by[session-tracked-orderedFruitsPaginated-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM
    (SELECT fruit.name AS name,
            fruit.id AS id
     FROM fruit AS fruit
     ORDER BY fruit.name ASC
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  ORDER BY fruit.name ASC
  LIMIT :param_3
  '''
# ---
# name: test_custom_query_hook_order_by[session-tracked-orderedFruitsPaginated-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM
    (SELECT fruit.name AS name,
            fruit.id AS id
     FROM fruit AS fruit
     ORDER BY fruit.name ASC
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  ORDER BY fruit.name ASC
  LIMIT :param_3
  '''
# ---
# name: test_custom_query_hook_where[session-tracked-filteredFruits-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  WHERE fruit.name = :name_1
  '''
# ---
# name: test_custom_query_hook_where[session-tracked-filteredFruits-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  WHERE fruit.name = :name_1
  '''
# ---
# name: test_custom_query_hook_where[session-tracked-filteredFruits-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM fruit AS fruit
  WHERE fruit.name = :name_1
  '''
# ---
# name: test_custom_query_hook_where[session-tracked-filteredFruitsPaginated-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM
    (SELECT fruit.name AS name,
            fruit.id AS id
     FROM fruit AS fruit
     WHERE fruit.name = :name_1
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  WHERE fruit.name = :name_2
  LIMIT :param_3
  '''
# ---
# name: test_custom_query_hook_where[session-tracked-filteredFruitsPaginated-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM
    (SELECT fruit.name AS name,
            fruit.id AS id
     FROM fruit AS fruit
     WHERE fruit.name = :name_1
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  WHERE fruit.name = :name_2
  LIMIT :param_3
  '''
# ---
# name: test_custom_query_hook_where[session-tracked-filteredFruitsPaginated-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.id
  FROM
    (SELECT fruit.name AS name,
            fruit.id AS id
     FROM fruit AS fruit
     WHERE fruit.name = :name_1
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  WHERE fruit.name = :name_2
  LIMIT :param_3
  '''
# ---
# name: test_load_columns_hook[session-tracked-fruits-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM fruit AS fruit
  '''
# ---
# name: test_load_columns_hook[session-tracked-fruits-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM fruit AS fruit
  '''
# ---
# name: test_load_columns_hook[session-tracked-fruits-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM fruit AS fruit
  '''
# ---
# name: test_load_columns_hook[session-tracked-fruitsPaginated-async-asyncpg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM
    (SELECT fruit.id AS id,
            fruit.name AS name,
            fruit.adjectives AS adjectives
     FROM fruit AS fruit
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  LIMIT :param_3
  '''
# ---
# name: test_load_columns_hook[session-tracked-fruitsPaginated-async-psycopg_async_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM
    (SELECT fruit.id AS id,
            fruit.name AS name,
            fruit.adjectives AS adjectives
     FROM fruit AS fruit
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  LIMIT :param_3
  '''
# ---
# name: test_load_columns_hook[session-tracked-fruitsPaginated-sync-psycopg_engine]
  '''
  SELECT fruit.name,
         fruit.adjectives,
         fruit.id
  FROM
    (SELECT fruit.id AS id,
            fruit.name AS name,
            fruit.adjectives AS adjectives
     FROM fruit AS fruit
     LIMIT :param_1
     OFFSET :param_2) AS fruit
  LIMIT :param_3
  '''
# ---
# name: test_query_hook_on_type[session-tracked-async-asyncpg_engine]
  '''
  SELECT color__fruits.name,
         color__fruits.id,
         color.id AS id_1
  FROM color AS color
  LEFT OUTER JOIN fruit AS color__fruits ON color.id = color__fruits.color_id
  WHERE color__fruits.name = :name_1
  '''
# ---
# name: test_query_hook_on_type[session-tracked-async-psycopg_async_engine]
  '''
  SELECT color__fruits.name,
         color__fruits.id,
         color.id AS id_1
  FROM color AS color
  LEFT OUTER JOIN fruit AS color__fruits ON color.id = color__fruits.color_id
  WHERE color__fruits.name = :name_1
  '''
# ---
# name: test_query_hook_on_type[session-tracked-sync-psycopg_engine]
  '''
  SELECT color__fruits.name,
         color__fruits.id,
         color.id AS id_1
  FROM color AS color
  LEFT OUTER JOIN fruit AS color__fruits ON color.id = color__fruits.color_id
  WHERE color__fruits.name = :name_1
  '''
# ---
